docs/ruby/README.md

```diff
@@ -0,0 +1,813 @@
+
+> [!NOTE]
+> Skip ahead if you want to build the application first and get into the details later. You won't need to
+> know about builders for the rest of this tutorial.
+
+In short, a builder is a delivery mechanism for buildpacks. A builder contains references to base images and individual buildpacks. A base image contains the operating system and system dependencies. Buildpacks are the components that will configure an image to run your application, that’s where the bulk of the logic lives and why the project is called “Cloud Native Buildpacks” and not “Cloud Native Builders.”
+
+You can view the contents of a builder via the command `pack builder inspect`. For example:
+
+```
+$ pack builder inspect heroku/builder:24 | grep Buildpacks: -m1 -A10
+Buildpacks:
+  ID                                NAME                               VERSION        HOMEPAGE
+  heroku/go                         Heroku Go                          0.4.1          https://github.com/heroku/buildpacks-go
+  heroku/gradle                     Heroku Gradle                      6.0.1          https://github.com/heroku/buildpacks-jvm
+  heroku/java                       Heroku Java                        6.0.1          https://github.com/heroku/buildpacks-jvm
+  heroku/jvm                        Heroku OpenJDK                     6.0.1          https://github.com/heroku/buildpacks-jvm
+  heroku/maven                      Heroku Maven                       6.0.1          https://github.com/heroku/buildpacks-jvm
+  heroku/nodejs                     Heroku Node.js                     3.2.9          https://github.com/heroku/buildpacks-nodejs
+  heroku/nodejs-corepack            Heroku Node.js Corepack            3.2.9          https://github.com/heroku/buildpacks-nodejs
+  heroku/nodejs-engine              Heroku Node.js Engine              3.2.9          https://github.com/heroku/buildpacks-nodejs
+  heroku/nodejs-npm-engine          Heroku Node.js npm Engine          3.2.9          https://github.com/heroku/buildpacks-nodejs
+```
+<!-- STOP. This document is autogenerated. Do not manually modify. See the top of the doc for more details. -->
+
+> [!NOTE]
+> Your output version numbers may differ.
+
+This output shows the various buildpacks that represent the different languages that are supported by this builder such as `heroku/go` and `heroku/nodejs-engine`.
+
+## Download an example Ruby on Rails application
+
+How do you configure a CNB? Give them an application. While Dockerfile is procedural, buildpacks, are declarative. A buildpack will determine what your application needs to function by inspecting the code on disk.
+
+For this example, we're using a pre-built Ruby on Rails application. Download it now:
+
+```
+$ git clone https://github.com/heroku/ruby-getting-started
+$ cd ruby-getting-started
+```
+<!-- STOP. This document is autogenerated. Do not manually modify. See the top of the doc for more details. -->
+
+Verify you're in the correct directory:
+
+```
+$ ls
```

Comment:
Should we always prefer `ls -1` like https://github.com/heroku/buildpacks/blob/f05e31822bceade3d7cb5136d72d0d0d80a9a87a/TUTORIAL.md?plain=1#L83. I like that it matches what customers would actually see, we could also go the other way, use formatting to simulate it being run in a TTY which gives us a shorter doc but is harder to skim at a glance.